runtimes/protocol/chat.ts

```diff
@@ -59,19 +62,23 @@ export interface QuickActionsOptions {
     quickActionsCommandGroups: QuickActionCommandGroup[]
 }
 
-export interface ChatRequest extends ChatParams {
-    partialResultToken?: ProgressToken
-}
-
-export const chatRequestType = new ProtocolRequestType<ChatRequest, ChatResult, ChatResult, void, void>(
-    CHAT_REQUEST_METHOD
-)
+export const chatRequestType = new AutoParameterStructuresProtocolRequestType<
+    ChatParams | EncryptedChatParams,
+    ChatResult | string,
+    ChatResult | string,
+    void,
+    void
+>(CHAT_REQUEST_METHOD)
 export const endChatRequestType = new ProtocolRequestType<EndChatParams, EndChatResult, never, void, void>(
     END_CHAT_REQUEST_METHOD
 )
-export const quickActionRequestType = new ProtocolRequestType<QuickActionParams, ChatResult, ChatResult, void, void>(
-    QUICK_ACTION_REQUEST_METHOD
-)
+export const quickActionRequestType = new AutoParameterStructuresProtocolRequestType<
+    QuickActionParams | EncryptedQuickActionParams,
```

Comment:
Do we have to change every chat request type to use these encrypted parameter versions or is this something we can solve at the transport layer?

Are these `Encrypted*` exposed to servers implementors?